<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from popsicle_bot.xacro             | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<!--
Popsicle stick robot
Copyright (C) 2015  Shane Loretz

This program is free software: you can redistribute it and/or modify
it under the terms of the GNU General Public License as published by
the Free Software Foundation, either version 3 of the License, or
(at your option) any later version.

This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.

You should have received a copy of the GNU General Public License
along with this program.  If not, see <http://www.gnu.org/licenses/>.
-->
<robot name="popsicle_bot" xmlns:xacro="http://www.ros.org/wiki/xacro">
  <link name="base_link">
    <visual>
      <geometry>
        <box size="0.2 0.28 0.06"/>
      </geometry>
      <origin xyz="0.1 -0.14 0.03"/>
      <material name="black">
        <color rgba="0.25 0.25 0.25 1"/>
      </material>
    </visual>
  </link>
  <joint name="base_to_main_stand" type="fixed">
    <parent link="base_link"/>
    <child link="main_stand"/>
    <origin xyz="0.051 -0.06 0.06"/>
  </joint>
  <link name="main_stand">
    <visual>
      <geometry>
        <box size="0.002 0.009 0.11"/>
      </geometry>
      <origin xyz="0 0 0.055"/>
      <material name="popsicle">
        <color rgba="0.9 0.9 0.5 1"/>
      </material>
    </visual>
  </link>
  <joint name="main_stand_to_motor_1" type="fixed">
    <parent link="main_stand"/>
    <child link="motor_1"/>
    <origin xyz="-0.001 -0.001 0.087"/>
  </joint>
  <link name="motor_1">
    <visual>
      <geometry>
        <mesh filename="package://popsicle_arm/urdf/Stepper_motor_28BYJ-48.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <origin rpy="0.0 1.57079632679 -1.57079632679" xyz="-0.019 0.024 0.021"/>
      <material name="motor">
        <color rgba="0.7 0.7 0.7 1"/>
      </material>
    </visual>
  </link>
  <joint name="motor_1_to_plastic_1" type="continuous">
    <parent link="motor_1"/>
    <child link="plastic_1"/>
    <axis xyz="1 0 0"/>
    <origin xyz="0.006 -0.008 0.0"/>
  </joint>
  <link name="plastic_1">
    <visual>
      <geometry>
        <box size="0.0027 0.038 0.023"/>
      </geometry>
      <origin xyz="0.0 -0.009 0"/>
      <material name="black">
        <color rgba="0.25 0.25 0.25 1"/>
      </material>
    </visual>
  </link>
  <joint name="plastic_1_to_m1_arm" type="fixed">
    <parent link="plastic_1"/>
    <child link="m1_arm"/>
    <origin xyz="0.00135 -0.011 -0.0115"/>
  </joint>
  <link name="m1_arm">
    <visual>
      <geometry>
        <box size="0.002 0.009 0.115"/>
      </geometry>
      <origin xyz="0 0 0.0575"/>
      <material name="popsicle">
        <color rgba="0.9 0.9 0.5 1"/>
      </material>
    </visual>
  </link>
  <joint name="m1_arm_to_motor_2" type="fixed">
    <parent link="m1_arm"/>
    <child link="motor_2"/>
    <origin xyz="-0.001 -0.001 0.092"/>
  </joint>
  <link name="motor_2">
    <visual>
      <geometry>
        <mesh filename="package://popsicle_arm/urdf/Stepper_motor_28BYJ-48.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <origin rpy="0.0 1.57079632679 -1.57079632679" xyz="-0.019 0.024 0.021"/>
      <material name="motor">
        <color rgba="0.7 0.7 0.7 1"/>
      </material>
    </visual>
  </link>
  <joint name="motor_2_to_plastic_2" type="continuous">
    <parent link="motor_2"/>
    <child link="plastic_2"/>
    <axis xyz="1 0 0"/>
    <origin xyz="0.006 -0.008 0.0"/>
  </joint>
  <link name="plastic_2">
    <visual>
      <geometry>
        <box size="0.0027 0.038 0.023"/>
      </geometry>
      <origin xyz="0.0 -0.009 0"/>
      <material name="black">
        <color rgba="0.25 0.25 0.25 1"/>
      </material>
    </visual>
  </link>
  <joint name="plastic_2_to_m2_arm" type="fixed">
    <parent link="plastic_2"/>
    <child link="m2_arm"/>
    <origin xyz="0.00135 -0.011 -0.0115"/>
  </joint>
  <link name="m2_arm">
    <visual>
      <geometry>
        <box size="0.002 0.009 0.115"/>
      </geometry>
      <origin xyz="0 0 0.0575"/>
      <material name="popsicle">
        <color rgba="0.9 0.9 0.5 1"/>
      </material>
    </visual>
  </link>
</robot>

